src/types/map.go

```diff
@@ -180,6 +188,36 @@ func (m Map) Equals(m2 Map) bool {
 	return true
 }
 
+// Union return new Map that contains all types from the current and other
+func (m Map) Union(other Map) Map {
```

Comment:
what `MergeMaps` didn't triple?

Reply:
Because we wanna union current with another vs Create new a lot of maps. Its ok to separate it

Reply:
I don't get it. We've created a new map here or there

Reply:
Union is logically tied to the current instance and merges a specific map with another, while MergeMaps can be used to merge any map